studio/components/interfaces/Billing/Billing.utils.ts

```diff
@@ -15,10 +16,22 @@ export const getProductPrice = (product: any) => {
   return price
 }
 
+// Products can have multiple prices, this is to get the correct one based on the subscription
+const getProductPriceId = (
+  currentSubscription: StripeSubscription,
+  product: any
+): string | undefined => {
+  const existingAddon = currentSubscription.addons.find((addon) => addon.prod_id === product.id)
+  const existingAddonPriceId = product.prices.find(
+    (price: any) => price.unit_amount === existingAddon?.unit_amount
```

Comment:
Is it correct that we find the existing add on based on the unit amount?

Reply:
yup! although not ideal - the API at the moment returns the addons within `currentSubscrition` without the price id, but only the unit_amount, hence why